aws_lambda_script/app/resources.py

```diff
@@ -0,0 +1,619 @@
+
+
+# Route to return the user email from the token in authorization header
+@ns.route("/user")
+class User(Resource):
+    @ns.doc(responses={200: "Success", 401: "Authorization is required"})
+    def get(self):
+        owner_email = get_user_email(parser.parse_args())
+        return {"email": owner_email}, 200
+
+
+# Route to return all projects with optional filters
+filterParser = reqparse.RequestParser()
+filterParser.add_argument(
+    "email",
+    location="args",
+    action="split",
+    required=False,
+    help="User email to filter by",
+)
+filterParser.add_argument(
+    "roles", location="args", action="split", required=False, help="Roles to filter by"
+)
+filterParser.add_argument(
+    "name",
+    location="args",
+    action="split",
+    required=False,
+    help="Project name to filter by",
+)
+filterParser.add_argument(
+    "developed",
+    location="args",
+    action="split",
+    required=False,
+    help="Developed partners to filter by",
+)
+filterParser.add_argument(
+    "languages",
+    location="args",
+    action="split",
+    required=False,
+    help="Languages to filter by",
+)
+filterParser.add_argument(
+    "source_control",
+    location="args",
+    action="split",
+    required=False,
+    help="Source control to filter by",
+)
+filterParser.add_argument(
+    "hosting",
+    location="args",
+    action="split",
+    required=False,
+    help="Hosting type to filter by",
+)
+filterParser.add_argument(
+    "database",
+    location="args",
+    action="split",
+    required=False,
+    help="Database to filter by",
+)
+filterParser.add_argument(
+    "frameworks",
+    location="args",
+    action="split",
+    required=False,
+    help="Frameworks to filter by",
+)
+filterParser.add_argument(
+    "CICD",
+    location="args",
+    action="split",
+    required=False,
+    help="CI/CD tools to filter by",
+)
+filterParser.add_argument(
+    "infrastructure",
+    location="args",
+    action="split",
+    required=False,
+    help="Infrastructure to filter by",
+)
+filterParser.add_argument(
+    "return",
+    location="args",
+    action="split",
+    required=False,
+    help="Sections to return: user, details, developed, source_control, architecture, or whole project",
+)
+
+filter_params = {
+    "email": "User email to filter by",
+    "roles": "Roles to filter by",
+    "name": "Project name to filter by",
+    "developed": 'Developed partners or "In house, partnership or outsourced" to filter by',
+    "languages": "Languages to filter by",
+    "source_control": "Source control to filter by",
+    "hosting": "Hosting type to filter by",
+    "database": "Database to filter by",
+    "frameworks": "Frameworks to filter by",
+    "CICD": "CI/CD tools to filter by",
+    "infrastructure": "Infrastructure to filter by",
+    "return": "Sections to return: user, details, developed, source_control, architecture, or whole project",
+}
+
+
+@ns.route("/projects/filter")
```

Comment:
This feels like it could be refactored. Is there a way to break up the loop through the various parameters into helpre functions or any duplicated code that could be made a utility function?

Reply:
Refactored with helper functions and reduced duplicate code. Filter isn't used anywhere but it's good to update it I guess.